pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3428,50 +3241,131 @@ var _ = Describe("Consolidation", func() {
-			labels := map[string]string{
-				"app": "test",
-			}
+		DescribeTable("can merge 3 nodes into 1",
+			func(spotToSpot bool) {
+				nodeClaims = lo.Ternary(spotToSpot, spotNodeClaims, nodeClaims)
+				nodes = lo.Ternary(spotToSpot, spotNodes, nodes)
+				// create our RS so we can link a pod to it
+				rs := test.ReplicaSet()
+				ExpectApplied(ctx, env.Client, rs)
+				pods := test.Pods(3, test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{Labels: labels,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion:         "apps/v1",
+								Kind:               "ReplicaSet",
+								Name:               rs.Name,
+								UID:                rs.UID,
+								Controller:         ptr.Bool(true),
+								BlockOwnerDeletion: ptr.Bool(true),
+							},
+						}}})
+
+				ExpectApplied(ctx, env.Client, rs, pods[0], pods[1], pods[2], nodeClaims[0], nodes[0], nodeClaims[1], nodes[1], nodeClaims[2], nodes[2], nodePool)
+				ExpectMakeNodesInitialized(ctx, env.Client, nodes[0], nodes[1], nodes[2])
+
+				// bind pods to nodes
+				ExpectManualBinding(ctx, env.Client, pods[0], nodes[0])
+				ExpectManualBinding(ctx, env.Client, pods[1], nodes[1])
+				ExpectManualBinding(ctx, env.Client, pods[2], nodes[2])
+
+				// inform cluster state about nodes and nodeclaims
+				ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{nodes[0], nodes[1], nodes[2]}, []*v1beta1.NodeClaim{nodeClaims[0], nodeClaims[1], nodeClaims[2]})
+
+				fakeClock.Step(10 * time.Minute)
+
+				var wg sync.WaitGroup
+				ExpectTriggerVerifyAction(&wg)
+				ExpectMakeNewNodeClaimsReady(ctx, env.Client, &wg, cluster, cloudProvider, 1)
+				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+				wg.Wait()
+
+				// Process the item so that the nodes can be deleted.
+				ExpectReconcileSucceeded(ctx, queue, types.NamespacedName{})
+
+				// Cascade any deletion of the nodeclaim to the node
+				ExpectNodeClaimsCascadeDeletion(ctx, env.Client, nodeClaims[0], nodeClaims[1], nodeClaims[2])
+
+				// three nodeclaims should be replaced with a single nodeclaim
+				Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(1))
+				Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(1))
+				ExpectNotFound(ctx, env.Client, nodeClaims[0], nodes[0], nodeClaims[1], nodes[1], nodeClaims[2], nodes[2])
+			},
+			Entry("if the candidate is on-demand node", false),
+			Entry("if the candidate is spot node", true),
+		)
+		It("can merge 3 nodes into 1 if the candidates have both spot and on-demand", func() {
+			// By default all the 3 nodeClaims are OD.
+			nodeClaims = lo.Ternary(false, spotNodeClaims, nodeClaims)
+			nodes = lo.Ternary(false, spotNodes, nodes)
+			// Change one of them to spot.
+			nodeClaims[2].Labels = lo.Assign(nodeClaims[0].Labels, map[string]string{
```

Comment:
shouldn't these indices match? probably doesn't make a difference though
```suggestion
			nodeClaims[2].Labels = lo.Assign(nodeClaims[2].Labels, map[string]string{
```

Reply:
Yes, it doesn't make a difference. So, I just chose the first one. I will change it if it is more readable.